src/core/cluster.py

```diff
@@ -124,42 +124,44 @@ def client_relations(self) -> set[Relation]:
         return set(self.model.relations[REL_NAME])
 
     @property
-    def peer_cluster_orchestrator_relations(self) -> set[Relation]:
-        """The `peer-cluster-orchestrator` relations that this charm is providing."""
-        return set(self.model.relations[PEER_CLUSTER_ORCHESTRATOR_RELATION])
+    def peer_cluster_orchestrator_relation(self) -> Relation | None:
+        """The `peer-cluster-orchestrator` relation that this charm is providing."""
+        return self.model.get_relation(PEER_CLUSTER_ORCHESTRATOR_RELATION)
 
     @property
     def peer_cluster_relation(self) -> Relation | None:
         """The `peer-cluster` relation that this charm is requiring."""
         return self.model.get_relation(PEER_CLUSTER_RELATION)
 
     @property
-    def peer_clusters(self) -> set[PeerCluster]:
-        """The state for all related `peer-cluster` applications that this charm is providing for."""
-        peer_clusters = set()
-        balancer_kwargs: dict[str, Any] = {
-            "balancer_username": self.cluster.balancer_username,
-            "balancer_password": self.cluster.balancer_password,
-            "balancer_uris": self.cluster.balancer_uris,
-        }
-        for relation in self.peer_cluster_orchestrator_relations:
-            if not relation.app or not self.runs_balancer:
-                continue
+    def peer_cluster_orchestrator(self) -> PeerCluster:
+        """The state for the related `peer-cluster-orchestrator` application that this charm is requiring from."""
+        balancer_kwargs: dict[str, Any] = (
+            {
+                "balancer_username": self.cluster.balancer_username,
+                "balancer_password": self.cluster.balancer_password,
+                "balancer_uris": self.cluster.balancer_uris,
+            }
+            if self.runs_balancer
+            else {}
+        )
 
-            peer_clusters.add(
-                PeerCluster(
-                    relation=relation,
-                    data_interface=PeerClusterOrchestratorData(self.model, relation.name),
-                    **balancer_kwargs,
-                )
-            )
+        return PeerCluster(
```

Comment:
**thought** one day we should probably provide an actual data model for these databags

Reply:
What do you mean?
We do have one (`PeerCluster`), it's just fairly complicated because of the super-application requiring all fields from 'somewhere'.